src/components/Pega_Extensions_BarCode/index.tsx

```diff
@@ -0,0 +1,45 @@
+import { Text, Card, CardHeader, CardContent } from '@pega/cosmos-react-core';
+import Barcode from 'react-barcode';
+import StyledWrapper from './styles';
+
+type BarCodeProps = {
+  heading: string;
+  format:
+    | 'CODE39'
+    | 'CODE128'
+    | 'EAN13'
+    | 'ITF14'
+    | 'MSI'
+    | 'pharmacode'
+    | 'codabar'
+    | 'upc'
+    | undefined;
+  selectionProperty: string;
+  displayValue: boolean;
+  getPConnect: any;
+};
+
+/*
+
+*/
+export default function PegaExtensionsBarCode(props: BarCodeProps) {
+  // eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars
+  const { heading = '', format, selectionProperty, displayValue, getPConnect } = props;
+  return (
+    <Card>
+      <CardHeader>
+        <Text variant='h2'>{heading}</Text>
+      </CardHeader>
+      <CardContent>
+        <StyledWrapper>
+          <Barcode
+            value={selectionProperty}
+            width={1}
```

Comment:
should the width be a parameter?

Reply:
I reverted it back to the default. This width represents the spacing between bars. Hence, when the value grows in length, there will be a horizontal layout shift. Though, this was the case when in widget mode – I did move it to field which is safe now

Reply:
It is recommended to stick with 2, using 1 it will make hard to scanner to catch it.